src/main/app/user_registration/ui.cljs

```diff
@@ -0,0 +1,81 @@
+        password-errors         (filter #(= :user-registration/password (:error/field-name %)) errors)
+        confirm-password-errors (filter #(= :user-registration/confirm-password (:error/field-name %)) errors)]
+    (dom/div
+      (dom/form {:onSubmit (fn [e] (.preventDefault e))}
+        (dom/div
+          (dom/input {:type     "text" :placeholder "Email"
+                      :onChange #(let [email (-> % .-target .-value)]
+                                   (comp/transact! this [(ur.mut/update-input
+                                                           {:user-registration/id    id
+                                                            :user-registration/email email})]))
+                      :onBlur   #(let [email (-> % .-target .-value)]
+                                   (comp/transact! this [(ur.mut/validate-input
+                                                          {:user-registration/id    id
+                                                           :user-registration/email email})]))})
+          (map #(ui-input-error %) email-errors))
+        (dom/div
+          (dom/input {:type     "password" :placeholder "Password"
+                      :onChange #(let [password (-> % .-target .-value)]
+                                   (comp/transact! this [(ur.mut/update-input
+                                                           {:user-registration/id       id
+                                                            :user-registration/password password})]))
+                      :onBlur   #(let [email (-> % .-target .-value)]
+                                   (comp/transact! this [(ur.mut/validate-input
+                                                           {:user-registration/id    id
+                                                            :user-registration/password email})]))
+                      })
+          (map #(ui-input-error %) password-errors))
+        (dom/div
+          (dom/input {:type     "password" :placeholder "Confirm password"
+                      :onChange #(let [confirm-password (-> % .-target .-value)]
+                                   (comp/transact! this [(ur.mut/update-input
+                                                           {:user-registration/id               id
+                                                            :user-registration/confirm-password confirm-password})]))
+                      :onBlur   #(let [email (-> % .-target .-value)]
+                                   (comp/transact! this [(ur.mut/validate-input
+                                                           {:user-registration/id    id
+                                                            :user-registration/confirm-password email})]))
+                      })
+          (map #(ui-input-error %) confirm-password-errors))
+        (dom/button {:disabled (or (seq errors) (not-every? #(or (not-empty %) (some? %)) (map identity [email password confirm-password])))
```

Comment:
Refactor - expression is too long. Additionally you can think of storing this information in app state.